lib/sgf/gtp_writer.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class SGF::GtpWriter < SGF::Writer
+  attr_writer :upside_down
+
+  def initialize
+    @upside_down = false
+  end
+
+  private
+
+  def gtp_move(node)
+    pps = node.properties
+    if pps['SZ']
+      @boardsize = pps['SZ'].to_i
+      out = []
+      out << "komi #{pps['KM']}" if pps['KM']
+      out << "boardsize #{pps['SZ']}\nclear_board"
+      pps['AB'].each {|pos| out << to_play("B", pos) } if pps['AB']
+      out.join("\n")
+    elsif pps.size > 1 || (pps.keys != ["B"] && pps.keys != ["W"])
+      ""
+    else
+      to_play(pps.keys.first, pps.values.first)
+    end
+  end
+
+  def to_play(color, pos)
+    if pos == ""
+      gtp_pos = "pass"
+    else
+      pos = pos.bytes
+      x = (pos[0] > 104 ? pos[0] + 1 : pos[0]).chr.upcase
```

Comment:
Isn't there a concern here that bytes aren't UTF-8 safe?

Reply:
not that I can think of, SGF positions are lowercase letters

Reply:
I'm still trying to understand why we're not using  string comparisons here - what's the need to go down to the level of bytes?

Reply:
I don't know how I would otherwise do math on chars...

Reply:
```
$ irb
2.5.0 :001 > "A"+1
Traceback (most recent call last):
        3: from /home/mathieu/.rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):1
        1: from (irb):1:in `+'
TypeError (no implicit conversion of Integer into String)
2.5.0 :002 > 
```

for some reason, GTP skip the letter `I` in the coordinate.
That's probably worth a comment

Reply:
You can do `"A".succ` to get the next letter :)

Reply:
supporting go board up to 25x25 size?
i don't know, this does what I want, I'm not sure how much more effort I want to put into this